frontend/packages/integration-tests-cypress/tests/crud/namespace-crud.spec.ts

```diff
@@ -53,54 +54,113 @@ describe('Namespace', () => {
     );
     nav.sidenav.clickNavLink(['Workloads', 'Secrets']);
+    listPage.rows.shouldBeLoaded();
     projectDropdown.shouldContain('All Projects');
     cy.log(
       'Nav back to list page from details page via breadcrumb should change Project back to "All Projects"',
     );
-    cy.get(`[data-test-rows="resource-row"]`)
-      .first()
-      .find('a')
-      .first()
-      .click();
+    listPage.rows.clickFirstLinkInFirstRow();
+    detailsPage.isLoaded();
     projectDropdown.shouldNotContain('All Projects'); // after drilldown to Details page, project should be specific to resource
-    cy.byLegacyTestID('breadcrumb-link-0').click();
+    detailsPage
+      .breadcrumb(0)
+      .contains('Secrets')
+      .click();
+    listPage.rows.shouldBeLoaded();
     projectDropdown.shouldContain('All Projects');
   });
 
   it('nav and breadcrumbs restores last selected Project when navigating from details to list view', () => {
     nav.sidenav.clickNavLink(['Workloads', 'Secrets']);
     projectDropdown.selectProject('default');
     projectDropdown.shouldContain('default');
-    cy.get(`[data-test-rows="resource-row"]`)
-      .first()
-      .find('a')
-      .first()
-      .click();
+    listPage.rows.clickFirstLinkInFirstRow();
+    detailsPage.isLoaded();
     projectDropdown.shouldContain('default');
     nav.sidenav.clickNavLink(['Workloads', 'Secrets']);
     projectDropdown.shouldContain('default');
-    cy.get(`[data-test-rows="resource-row"]`)
-      .first()
-      .find('a')
-      .first()
-      .click();
+    listPage.rows.clickFirstLinkInFirstRow();
+    detailsPage.isLoaded();
     projectDropdown.shouldContain('default');
-    cy.byLegacyTestID('breadcrumb-link-0').click();
+    detailsPage
+      .breadcrumb(0)
+      .contains('Secrets')
+      .click();
+    listPage.rows.shouldBeLoaded();
     projectDropdown.shouldContain('default');
   });
+
+  ['Roles', 'RoleBindings'].forEach((rolesOrBindings) => {
+    it(`tests ${rolesOrBindings} and Cluster${rolesOrBindings} detail page breadcrumbs`, () => {
+      nav.sidenav.clickNavLink(['User Management', rolesOrBindings]);
+
+      cy.log(`test ${rolesOrBindings} & All Projects`);
+      projectDropdown.selectProject('All Projects');
+      listPage.filter.by('cluster');
+      listPage.filter.numberOfActiveFiltersShouldBe(1);
+      listPage.rows.clickFirstLinkInFirstRow();
+      detailsPage.isLoaded();
+      projectDropdown.shouldNotExist();
+      detailsPage
+        .breadcrumb(0)
+        .contains(rolesOrBindings)
+        .click();
+      listPage.rows.shouldBeLoaded();
+      projectDropdown.shouldContain('All Projects');
+      listPage.titleShouldHaveText(rolesOrBindings);
+
+      cy.log(`test Cluster${rolesOrBindings} & openshift-console project`);
+      projectDropdown.selectProject('openshift-console');
+      projectDropdown.shouldContain('openshift-console');
```

Comment:
`default` would be better so the tests run on native k8s where this project won't exist.

```suggestion
      projectDropdown.selectProject('default');
      projectDropdown.shouldContain('default');
```

Reply:
We refer to the project name enough times it would be good to assign it to a const.

Reply:
fixed